torch/csrc/jit/script/init.cpp

```diff
@@ -208,7 +208,7 @@ c10::optional<IValue> tryCalculateDefaultParam(
     } else {
       return toIValue(def_value, arg.type());
     }
-  } catch (py::cast_error& e) {
+  } catch (...) {
```

Comment:
It's still throwing a pybind error right? It looks like they all subclass `builtin_exception` 

https://github.com/pybind/pybind11/blob/master/include/pybind11/detail/common.h#L664

We don't want to catch assert failures from our own code in `toIValue`.

Reply:
We've gone through this in the past with constant propagation. At first we tried to only catch errors that were thrown by us, then all sorts of weird exceptions propagated, and then we caught all exceptions. We don't know what weird exceptions could be thrown, in any case the `Expected a default value of type for parameter ....` will probably be better.

Reply:
We should just have an error free `tryToIValue` that this should use, but `...` is fine for now